File: pinyougou/pinyougou-manage-web/src/main/webapp/js/service/goodsService.js
//定义业务服务
app.service("goodsService",function ($http) {
    //加载列表数据
    this.findAll = function(){
        return $http.get("../goods/findAll.do");
    };

    this.findPage = function (page, rows) {
        return $http.get("../goods/findPage.do?page=" + page + "&rows=" + rows);
    };

    this.add = function (entity) {
        return $http.post("../goods/add.do",entity);
    };

    this.update = function (entity) {
        return $http.post("../goods/update.do",entity);
    };

    this.findOne = function (id) {
        return $http.get("../goods/findOne.do?id=" + id);
    };

    this.delete = function (selectedIds) {
        return $http.get("../goods/delete.do?ids=" + selectedIds);
    };

    this.search = function (page, rows, searchEntity) {
        return $http.post("../goods/search.do?page=" + page + "&rows=" + rows, searchEntity);

    };

    this.updateStatus = function (selectedIds, status) {
        return $http.get("../goods/updateStatus.do?ids=" + selectedIds + "&status=" + status);
    };
});

File: pinyougou/pinyougou-sellergoods/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/GoodsService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbGoods;
import com.pinyougou.pojo.TbItem;
import com.pinyougou.service.BaseService;
import com.pinyougou.vo.Goods;
import com.pinyougou.vo.PageResult;

import java.util.List;

public interface GoodsService extends BaseService<TbGoods> {
    PageResult search(TbGoods goods, Integer page, Integer rows);

    void addGoods(Goods goods);

    Goods findGoodsById(Long id);

    void updateGoods(Goods goods);

    void updateStatus(Long[] ids, String status);

    void deleteGoodsByIds(Long[] ids);


    void itemUpshelf(Long[] ids);

    void soldOut(Long[] ids);

    /**
     * 根据商品SPU id集合和状态查询这些商品对应的SKU列表
     * @param ids 商品SPU id集合
     * @param status sku商品状态 上下架状态
     * @return  sku商品列表
     */
    List<TbItem> findItemListByGoodsIdsAndStatus(Long[] ids, String status);

    //根据商品id查询商品基本,描述,启用的sku列表
    Goods findGoodsByIdAndStatus(Long goodsId, String s);

}
